apps/files/src/components/FileList.vue

```diff
@@ -1,89 +1,98 @@
-              <oc-table-cell shrink type="head">
-                <oc-checkbox class="uk-margin-small-left" id="filelist-check-all" @click.native="toggleAll" :value="selectedAll" />
-              </oc-table-cell>
-              <oc-table-cell shrink type="head" v-if="!publicPage()" />
-              <oc-table-cell type="head" class="uk-text-truncate" v-translate>Name</oc-table-cell>
-              <oc-table-cell shrink type="head" :class="{ 'uk-visible@s' : !_sidebarOpen, 'uk-visible@m'  : _sidebarOpen }"><translate>Size</translate></oc-table-cell>
-              <oc-table-cell shrink type="head" :class="{ 'uk-visible@s' : !_sidebarOpen, 'uk-visible@m'  : _sidebarOpen }" class="uk-text-nowrap" v-translate>Modification Time</oc-table-cell>
-              <oc-table-cell shrink type="head" :class="{ 'uk-visible@s' : _sidebarOpen }" v-translate>Actions</oc-table-cell>
-            </oc-table-row>
-          </thead>
-          <oc-table-group>
-            <oc-table-row v-for="(item, index) in fileData" :key="index" :class="_rowClasses(item)" @click="selectRow(item, $event)" :id="'file-row-' + item.id">
-              <oc-table-cell>
-                <oc-checkbox class="uk-margin-small-left" @click.stop @change.native="$emit('toggle', item)" :value="selectedFiles.indexOf(item) >= 0" />
-              </oc-table-cell>
-              <oc-table-cell class="uk-padding-remove" v-if="!publicPage()">
-                <oc-star class="uk-display-block" @click.native.stop="toggleFileFavorite(item)" :shining="item.starred" />
-              </oc-table-cell>
-              <oc-table-cell class="uk-text-truncate">
-                <oc-file @click.native.stop="item.type === 'folder' ? navigateTo(item.path.substr(1)) : openFileActionBar(item)"
-                        :name="$_ocFileName(item)" :extension="item.extension" class="file-row-name" :icon="fileTypeIcon(item)"
-                        :filename="item.name" :key="item.id"/>
-                <oc-spinner
-                  v-if="$_actionInProgress(item)"
-                  size="small"
-                  :uk-tooltip="$_disabledActionTooltip(item)"
-                  class="uk-margin-small-left"
-                />
-              </oc-table-cell>
-              <oc-table-cell class="uk-text-meta uk-text-nowrap" :class="{ 'uk-visible@s' : !_sidebarOpen, 'uk-visible@m'  : _sidebarOpen }">
-                {{ item.size | fileSize }}
-              </oc-table-cell>
-              <oc-table-cell class="uk-text-meta uk-text-nowrap" :class="{ 'uk-visible@s' : !_sidebarOpen, 'uk-visible@m'  : _sidebarOpen }">
-                {{ formDateFromNow(item.mdate) }}
-              </oc-table-cell>
-              <oc-table-cell :class="{ 'uk-visible@s' : _sidebarOpen }" class="uk-position-relative">
-                <div class="uk-button-group uk-margin-small-right" :class="{ 'uk-visible@m' : !_sidebarOpen, 'uk-visible@xl' : _sidebarOpen  }">
+      <virtual-scroll
+        v-if="fileData.length"
```

Comment:
This condition needs to be there. Otherwise, the virtual scroller would get initialised with empty array and to display items user would have to scroll first.